sdk/identity/src/client_credentials_flow/mod.rs

```diff
@@ -61,45 +62,26 @@ pub async fn perform(
         "https://login.microsoftonline.com/{}/oauth2/v2.0/token",
         tenant_id
     ))
-    .map_err(|_| ClientCredentialError::InvalidTenantId(tenant_id.to_owned()))?;
+    .with_context(ErrorKind::Credential, || {
+        format!("The supplied tenant id could not be url encoded: {tenant_id}")
+    })?;
 
     let response = client
         .post(url)
         .header("ContentType", "Application / WwwFormUrlEncoded")
         .body(encoded)
         .send()
         .await
-        .map_err(|e| ClientCredentialError::Request(Box::new(e)))?;
+        .map_kind(ErrorKind::Credential)?;
```

Comment:
Perhaps this should be `ErrorKind::Io` if it fails to send?

Reply:
Yeah, I wasn't sure if everything should wrap into `ErrorKind::Credential` or not in `azure_identity` crate. May be not.